### PR TITLE
Add pendulum 3 support

### PR DESCRIPTION
**Pull request by gibsondan**
Summary:
pendulum 3 didn't have any truly breaking changes so this is as simple as inverting our version check (which is probably how it should have worked originally). I think testing all three versions is warranted given how critical the package working correctly is for things like DST transitions and AMP working correctly.

Test Plan: BK

### CODE REVIEW

**Comment by gibsondan** [comment]
hm test failures suggest there may be more to be done here

**Comment by OwenKephart** [comment]
@gibsondan https://pendulum.eustace.io/blog/announcing-pendulum-3-0-0.html mentions that `Period` has been renamed `Interval` 😵

File: python_modules/dagster/dagster/_seven/compat/pendulum.py [hunk 1]
```diff
@@ -60,6 +60,9 @@ def create_pendulum_time(year, month, day, *args, **kwargs):
 )
 
 
+pendulum_time_travel = pendulum.travel_to if _IS_PENDULUM_3 else pendulum.time
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/_seven/compat/pendulum.py:63]: not to be too pedantic, but this is actually subtly different I believe.

`pendulum.travel_to` keeps the clock ticking, whereas `pendulum.time` freezes the time entirely. so think actually you'd want lambda t: pendulum.travel_to(t, freeze=true) here

Comment (OwenKephart) [reply python_modules/dagster/dagster/_seven/compat/pendulum.py:63]: with that in mind, I think this should be called pendulum_freeze_time, as I don't think the older versions actually support the non-frozen behavior

**Pull request closed without merging**